src/agent/debugger/src/target.rs

```diff
@@ -150,17 +212,24 @@ impl Target {
         self.current_thread_handle
     }
 
+    pub fn current_thread_id(&self) -> DWORD {
+        self.current_thread_id
+    }
+
     pub fn create_new_thread(&mut self, thread_handle: HANDLE, thread_id: DWORD) {
         self.current_thread_handle = thread_handle;
-        self.thread_handles.insert(thread_id, thread_handle);
+        self.thread_info.insert(
+            thread_id,
+            ThreadInfo::new(thread_id, thread_handle, self.wow64),
+        );
     }
 
     pub fn set_current_thread(&mut self, thread_id: DWORD) {
-        self.current_thread_handle = *self.thread_handles.get(&thread_id).unwrap();
+        self.current_thread_handle = self.thread_info.get(&thread_id).unwrap().handle;
```

Comment:
Not new, but should we really be `unwrap()`-ing the `get()`, here?

Reply:
It does seem panic worthy, as in we or the the Win32 debug api has a serious logic error, but I'll consider adding a log message in another PR.